pkg/sql/pgwire/testdata/pgtest/enum

```diff
@@ -46,14 +46,64 @@ RowDescription
 # DataTypeSize for an enum is 4, as floats are used to represent enums
 # internally (4 bytes). Since our encodings are variable size, we report
 # the DataTypeSize to be -1, which is the variable length size.
-until ignore_type_oids crdb_only
+until crdb_only
 RowDescription
 ----
-{"Type":"RowDescription","Fields":[{"Name":"te","TableOID":0,"TableAttributeNumber":0,"DataTypeOID":0,"DataTypeSize":-1,"TypeModifier":-1,"Format":0}]}
+{"Type":"RowDescription","Fields":[{"Name":"te","TableOID":0,"TableAttributeNumber":0,"DataTypeOID":100052,"DataTypeSize":-1,"TypeModifier":-1,"Format":0}]}
 
 until
 ReadyForQuery
 ----
 {"Type":"DataRow","Values":[{"text":"hi"}]}
 {"Type":"CommandComplete","CommandTag":"SELECT 1"}
 {"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Regression for #53413. This test ensures that the wire protocol can handle
+# user defined type OIDs.
+
+send
+Query {"String": "CREATE TABLE tb (x te)"}
+----
+
+until
+ReadyForQuery
+----
+{"Type":"CommandComplete","CommandTag":"CREATE TABLE"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Prepare a query and type hint a user defined type. Then bind this prepared
+# statement with a user defined type argument ([104, 105] = 'hi').
+send
+Parse {"Name": "s1", "Query": "INSERT INTO tb VALUES ($1)", "ParameterOIDs": [100052]}
+Bind {"DestinationPortal": "p", "PreparedStatement": "s1", "ParameterFormatCodes": [0], "Parameters": [[104, 105]]}
+Execute {"Portal": "p"}
+Sync
+----
+
+until
+ReadyForQuery
+----
+{"Type":"ParseComplete"}
+{"Type":"BindComplete"}
+{"Type":"CommandComplete","CommandTag":"INSERT 0 1"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Ensure that our value was successfully inserted.
+send
+Query {"String": "SELECT * FROM tb"}
```

Comment:
would it be worth testing SELECT with a prepared statement too? up to you -- i'm not sure if the code paths are significantly different.

Reply:
I don't think that's needed, they both go through the same path here